### ISSUE TITLE
Integrate replicate with Hugging Face transformers

### ISSUE THREAD

**Issue by charlieyou** [replicate/replicate#498]
Similar to Pytorch Lightning, Hugging Face abstracts away the training loop and provides [a Trainer API](https://huggingface.co/transformers/main_classes/trainer.html) with [callbacks](https://huggingface.co/transformers/main_classes/callback.html#transformers.TrainerCallback)

I will probably be able to take this on in the next few weeks.

How are you thinking about keeping callbacks in this repo (for both this and previous ones (keras, PL)) vs contributing to those library repos like tensorboard and w&b do? Keep in here until a stable v1 release?

**Comment by bfirsh** [comment]
Nice, great idea! 🙌

We're going to contribute the PyTorch Lightning integration back to PyTorch Lightning. We included in our repository as a shortcut, because the PyTorch lightning needed changes to their API.

Looks like Hugging Face also include callbacks for services in their library, so it probably makes sense for it to live there. As you suggest though, our API is unstable, so maybe it makes sense to not include it right now.

As a place to develop it, I wonder if it makes sense to create a separate package? `replicate-huggingface`, or whatever. The intention would be to get it into Hugging Face, and the advantage of it being separate is it isn't something we have to _remove_ from Replicate at some point, which is always messy. We can document it or link to it from https://replicate.ai so it gets eyeballs on it.

**Comment by bfirsh** [comment]
@srush Do you have any opinions about this, or know someone at Hugging Face who might? :)

**Comment by srush** [comment]
I will check!

**Comment by MusicStudioNYC** [comment]
Bumping this. Is there a streamlined process to migrate a huggingface model to run on Replicate?
